Correct TVM contact constraints to enable torque control

Hugo-L3174 opened the pull request:
This correctly implements velocity and acceleration contact constraints for TVM: only position was used before, regardless of the contact constraint type, which made torque control fail.

- The solver contact constraint now passes the contact type to the tvm backend instead of only to tasks.
- This type is stored in the TVMQPSolver when a contact type constraint is added to the solver and is used to implement the correct type when a new contact is added.

Hugo-L3174 committed:
Correct contact dynamics + loading structure for tvm

ThomasDuvinage committed:
feat(contactConstraint): use existing contact type and rename setter

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ThomasDuvinage committed:
feat(contactConstraint): avoid undefined behaivor on constrait type if constraint is created but not added to solver

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

arntanguy reviewed (changes requested):
Thanks @Hugo-L3174 for noticing this and proposing a fix.

Some comments inline, essentially I think we shouldn't duplicate the contact type logic in both the `ContactConstraint` and `TVMQPSolver`. Also please use enum types instead of int ;)

Other than that, not much to add within the scope of this PR. Although the more I look at the current contact handling logic, the more convinced I become that a serious refactoring would be needed, but that's definitely for another time.

Hugo-L3174 committed:
Set contact type as enum class + avoid storing type twice

- Type is set as enum class instead of plain to avoid potential issues.
- TVMQPSolver now casts the ContactConstraint directly from the constraints set as suggested, avoiding to store type both in solver and constraint

Removing the contact constraint during control will not update the already present contacts but this might be better addressed in a contact constraint overhaul given that it is not common use.

Hugo-L3174 committed:
fix: unnecessary include + safety loop break
